src/popup/components/TransferReview.vue

```diff
@@ -83,178 +83,151 @@
   aeToAettos,
   checkAensName,
   convertToken,
   escapeSpecialChars,
 } from '../utils';
+import { IAccount } from '../../types';
 import { MODAL_DEFAULT, AETERNITY_CONTRACT_ID } from '../utils/constants';
 import DetailsItem from './DetailsItem.vue';
 import TokenAmount from './TokenAmount.vue';
 import AvatarWithChainName from './AvatarWithChainName.vue';
 import ModalHeader from './ModalHeader.vue';
 
-export default {
+export default defineComponent({
   name: 'TransferReview',
   components: {
     ModalHeader,
     AvatarWithChainName,
     DetailsItem,
     TokenAmount,
   },
-  mixins: [deeplinkApi],
   model: {
     prop: 'transferData',
   },
   props: {
     transferData: { type: Object, required: true },
     isAddressChain: Boolean,
     isAddressUrl: Boolean,
+    recipientAddress: { type: String, required: false, default: null },
+    amount: { type: Number, required: false, default: null },
   },
-  data() {
-    return {
-      loading: false,
-      AETERNITY_CONTRACT_ID,
-    };
-  },
-  computed: {
-    ...mapGetters(['account']),
-    ...mapState([
-      'sdk',
-      'tippingV1',
-      'tippingV2',
-    ]),
-    isSelectedAssetAex9() {
-      return !!this.transferData.selectedAsset
-        && this.transferData.selectedAsset.contractId !== AETERNITY_CONTRACT_ID;
-    },
-    isRecipientName() {
-      return this?.recipientAddress && checkAensName(this.recipientAddress);
-    },
-    tokenSymbol() {
-      return this.transferData.selectedAsset?.symbol || '-';
-    },
-    tippingContract() {
-      return this.tippingV2 || this.tippingV1;
-    },
-  },
-  methods: {
-    async submit() {
-      const {
-        amount: amountRaw,
-        address: recipient,
-        selectedAsset,
-        note,
-      } = this.transferData;
+  setup(props, { root, emit }) {
+    const { openCallbackOrGoHome } = useDeepLinkApi({ router: root.$router });
+    const loading = ref<boolean>(false);
+    const account = computed<IAccount>(() => root.$store.getters.account);
+    const tippingV1 = computed(() => root.$store.state.tippingV1);
+    const tippingV2 = computed(() => root.$store.state.tippingV2);
+    const sdk = computed(() => root.$store.state.sdk);
+    const isRecipientName = computed(
+      () => props.recipientAddress && checkAensName(props.recipientAddress),
+    );
+    const tokenSymbol = computed(() => props.transferData.selectedAsset?.symbol || '-');
+    const tippingContract = computed(() => tippingV2.value || tippingV1.value);
 
-      if (!amountRaw || !recipient || !selectedAsset) {
-        return;
-      }
+    const isSelectedAssetAex9 = computed(() => (
+      !!props.transferData.selectedAsset
+        && props.transferData.selectedAsset.contractId !== AETERNITY_CONTRACT_ID
+    ));
 
-      const amount = (selectedAsset.contractId === AETERNITY_CONTRACT_ID)
-        ? aeToAettos(amountRaw)
-        : convertToken(amountRaw, selectedAsset.decimals);
+    const openTransactionFailedModal = () => {
```

Comment:
This is rather subjective, but there is a nice pattern to make the `setup` hook contents more readable: to create methods as a regular functions, not a constants with arrow function. It makes easier to distinguish the methods from the data.
Eg.:
```
const foo = computed(() => ...);
const bar = ref(false);

function someMethod() { ... }

function anotherMethod() { ... }
```